test/raft/unit/test_snapshot.c

```diff
@@ -287,10 +318,12 @@ TEST(snapshot_leader, basic, set_up, tear_down, 0, NULL) {
 	ut_disk_io_done(&leader.work);
```

Comment:
Seems strange to run `_io()` and `io_done()` callbacks twice on the same work item, without `work_init()` and `work_queue()` in the middle.

Reply:
We discussed this offline, it happens because the leader needs to:
1. read a page from its local db file and get its checksum,
2. lookup the checksum in the hashtable that the follower sent.

Because these are sequential operations it is simpler to schedule the second when the first one is finished. It happens in the following way:
1. disk_io calls the work_cb itself,
2. disk_io_done calls the after_work_cb which internally calls leader_tick(M_WORK_DONE),
3. inside leader tick we move to the next state which schedules another background work,
4. disk_io ...
5. disk_io_done ...
6. leader_tick goes into the next state that sends the message.